service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/BaseDevice.java

```diff
@@ -60,7 +61,7 @@ protected BaseDevice(String deviceId, SymmetricKey symmetricKey)
      * @param deviceId Name of the device (used as device id)
      * @param authenticationType - The type of authentication used by this device.
      */
-    protected BaseDevice(String deviceId, AuthenticationType authenticationType)
+    BaseDevice(String deviceId, AuthenticationType authenticationType)
```

Comment:
Does this need the private keyword like other places?

Reply:
This constructor is called from another class within this package, so I can't mark it as private. This lack of modifier makes the constructor "Package private" which means only classes within this package can access it. Classes outside of the package, even if they extend this class, can't access it